pkg/virt-handler/vm.go

```diff
@@ -2794,6 +2811,17 @@ func (d *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
 		if err := d.getMemoryDump(vmi); err != nil {
 			return err
 		}
+
+		if err := d.hotplugVirtioInterfaces(vmi); err != nil {
+			log.Log.Object(vmi).Error(err.Error())
+			var criticalNetErr *neterrors.CriticalNetworkError
+			if !goerror.As(err, &criticalNetErr) {
+				log.Log.Object(vmi).Infof("pod interface not ready yet; re-queuing VMI reconcile operation to give it more time")
+				d.Queue.AddAfter(controller.VirtualMachineInstanceKey(vmi), 200*time.Millisecond)
```

Comment:
@maiqueb  there's no way to prevent, ipam=false to continue, so it will missing `-nic` interface still.

```go
if len(addrList) == 0 {
		b.ipamEnabled = false
```
this check may not reliable, since ip address may added after the `BridgePodNetworkConfigurator) DiscoverPodNetworkInterface` function call.

my previous `preCheck` is not reliable either, it just got lucky with 1 second interval (200ms have no that lucky anymore), though depends on the timing is not a good idea.

maybe we need some other way, I see you previously check multus status from pod annotation? ( only if multus status is ready then move on to plugphase1? )

Reply:
Yes, I recently became aware of the race between CNI IPAM and the virt-handler checking the link's state.

Unfortunately, I think the safest course of action is to recover some of the state I had previously; plan is for virt-controller to patch the status of the VMI when it sees the pod features a CNI result for the hotplugged interfaces, and for virt-handler to hang with the hotplug operation until the vmi.status.interface.hotplug features the correct state (some indication the pod link is ready).